hasher-matcher-actioner/hmalib/lambdas/api/matches.py

```diff
@@ -240,4 +282,32 @@ def request_signal_opinion_change() -> ChangeSignalOpinionResponse:
 
         return ChangeSignalOpinionResponse(success)
 
+    @matches_api.get("/for-hash/", apply=[jsoninator(MatchesForHashRequest)])
+    def for_hash(request: MatchesForHashRequest) -> MatchesForHashResponse:
+        """
+        For a given hash/signal check the index(es) for matches and return the details
+        """
+        signal_type = None
+        if request.signal_type == "pdq":
+            # todo translate in MatchesForHashRequest and extend to cover MD5
+            signal_type = PdqSignal
+
+        if not signal_type:
+            # only support PDQ at the moment
+            bottle.response.status = 400
+            return MatchesForHashResponse([], request.signal_value)
+
+        matches = _get_matcher(indexes_bucket_name).match(
+            signal_type, request.signal_value
+        )
+
+        match_objects = []
+
+        for match in matches:
+            match_objects.extend(
+                Matcher.get_metadata_objects_from_match(signal_type, match)
```

Comment:
AFAIK, this will not write signal metadata. Is that a conscious choice?

Reply:
for now yes, long term we probably will want to but for the time being I'm preferring to leave hitting this endpoint side effect free.